### PR TITLE
[feat] add make migrations

### LINKED ISSUE #308: Create target `make migrations`
https://github.com/zenstruck/foundry/pull/306#issuecomment-1270237991

### PR DESCRIPTION
**Pull request by nikophil**
fixes #308

added these make target:
```shell
database-generate-migration    Generate new migration based on mapping in Zenstruck\Foundry\Tests\Fixtures\Entity
database-validate-mapping      Validate mapping in Zenstruck\Foundry\Tests\Fixtures\Entity
database-drop-schema           Drop database schema
```

### CODE REVIEW

File: bin/doctrine [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env php
```
Comment (nikophil) [inline bin/doctrine:1]: this file is needed to validate the db schema

https://www.doctrine-project.org/projects/doctrine-orm/en/current/reference/tools.html#doctrine-console

File: config/cli-config.php [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<?php
```
Comment (nikophil) [inline config/cli-config.php:1]: this file is needed to tell to doctrine/migrations which entity maanger to use. Its name is hard coded in migrations

File: tests/Fixtures/Entity/Comment.php [hunk 3]
```diff
@@ -18,7 +18,7 @@ class Comment
     private $id;
 
     /**
-     * @ORM\ManyToOne(targetEntity=User::class)
+     * @ORM\ManyToOne(targetEntity=User::class, inversedBy="comments")
```
Comment (nikophil) [inline tests/Fixtures/Entity/Comment.php:21]: `make migrations-validate` indicated me that there was a problem in the mapping :)

File: tests/Fixtures/Migrations/Version20221010154036.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    {
+        return 'Sixth migration.';
+    }
+
+    public function up(Schema $schema): void
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->addSql('ALTER TABLE posts CHANGE type type VARCHAR(255) NOT NULL');
+        $this->addSql('ALTER TABLE posts RENAME INDEX idx_985dbafad126f52 TO IDX_885DBAFA20DBE482');
+    }
```
Comment (nikophil) [inline tests/Fixtures/Migrations/Version20221010154036.php:25]: it seems that this was missing in the previous migrations

File: Makefile [hunk 5]
```diff
@@ -79,6 +79,20 @@ sca: docker-start bin/tools/psalm/vendor ### Run Psalm
 bin/tools/psalm/vendor: vendor bin/tools/psalm/composer.json bin/tools/psalm/composer.lock
 	@${DOCKER_PHP} composer bin psalm update
 
+database-generate-migration: docker-start vendor ### Generate new migration based on mapping in Zenstruck\Foundry\Tests\Fixtures\Entity
+	@${DC_EXEC} -e DATABASE_URL=${MYSQL_URL} php vendor/bin/doctrine-migrations migrations:migrate --no-interaction --allow-no-migration # first, let's load into db existing migrations
+	@${DC_EXEC} -e DATABASE_URL=${MYSQL_URL} php vendor/bin/doctrine-migrations migrations:diff --no-interaction
+	@${DC_EXEC} -e DATABASE_URL=${MYSQL_URL} php vendor/bin/doctrine-migrations migrations:migrate --no-interaction # load the new migration
+
+database-validate-mapping: docker-start vendor database-drop-schema ### Validate mapping in Zenstruck\Foundry\Tests\Fixtures\Entity
```
Comment (nikophil) [inline Makefile:87]: we must drop the schema before validating it, thjis way we're also validating our migrations

**Review by kbond — APPROVED** [review]
LGTM!

**Pull request merged**